---
title: "CNHplus: the chromosomal copy number heterogeneity which respects biological constraints"
author: "Marian GrendÃ¡r"
date: "`r format(Sys.Date(), '%d/%m/%Y')`"
output: rmarkdown::html_vignette
bibliography: refs.bib
vignette: >
  %\VignetteIndexEntry{CNHplus: the chromosomal copy number heterogeneity which respects biological constraints}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

<style type="text/css">
  body{
  font-size: 12pt;
   font-family: "Times New Roman", Times, serif;
}
</style>

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Introduction

_CNHplus_ provides R functions for estimating the tumor purity, the tumor ploidy, the Absolute Copy Number (ACN) profile of a tumor and the Copy Number Heterogeneity (CNH) [@van2021chromosomal] from the Relative Copy Number (RCN) profile of a sample, by the CNHplus method [@grendar2022cnhplus]. CNHplus modifies the original CNH method [@van2021chromosomal] by imposing the non-negativity constraint on the recovered ACN profile.


## CNH and CNH+

[@van2021chromosomal] introduced a single-sample method for estimating the Intra-Tumor Heterogeneity from Copy Number Variation (CNV) data, dubbed Copy Number Heterogeneity. The ACN profiles, recovered by CNH, may contain segments with the negative number of copies; cf. [@grendar2022cnhplus].
CNHplus method, - CNH+,  for short, -  implemented in _CNHplus_ library, corrects this omission in CNH as it imposes the non-negativity constraint on ACN of tumor in searching for the solution of the optimization problem that defines CNH.


## Loading CNHplus

_CNHplus_ and other libraries necessary for illustrating its use should be loaded, at first.


```{r message = FALSE}
library('CNHplus')
library('TCGAbiolinks')
library('stringi')
library('openxlsx')
library('dplyr')
```


## Input data 

The RCN profile of a sample, which is the input to CNH computations, can be obtained either by the microarray technology or by shallow Whole Genome Sequencing. The RCN profile data frame should comprise the following variables, at minimum:

* `Sample`
* `Chromosome`
* `Start`
* `End`
* `Segment_Mean`

Note that _CNHplus_ assumes that Segment_Mean is the relative copy number of the segment, not its log2 transformation.

## Plotting RCN profile of a sample

In order to reproduce the example from [@grendar2022cnhplus], it is necessary to load the RCN data for the TCGA-AJ-A3EJ-01A-11D-A19X-01 sample, which are included in _CNHplus_.

```{r}
data(sample_data)
head(sample_data)
```

Next, in order to plot the RCN profile of the sample, it is necessary to prepare the RCN profile (by anti-log2 transforming the Segment_Mean data) and a vector with the widths of the segments.

```{r}
sa = sample_data
sample_name = sa$Sample[1]
r = 2^sa$Segment_Mean
w = sa$End - sa$Start # should be +1
```

`plot_profile` function then plots the RCN profile of the sample and saves it to a tiff file with file name specified by `sample_name`.

```{r, fig.width=8, fig.height=4, include=T, fig.align='center', fig.show='hold', results=FALSE}
# RCN profile plot
plot_profile(sa, r, paste0(sample_name, '_RCN.tiff'),
             ylim = c(0,5), ylab = 'RCN')
```

The function can be used also for plotting the ACN profiles.


## Finding CNH+ (and CNH)

_CNHplus_ finds a solution of the optimization problem (4') from [@grendar2022cnhplus]
by searching over a grid of pairs of tumor purities and tumor ploidies. The grid can be created by `make_grid` function.

```{r}
# making grid
grid = make_grid(purity = seq(0.2, 1, 0.01), ploidy = seq(1.5, 5, 0.01))
```

`find_cnhplus` function requires the grid and relative copy number profile as input.
The number of candidate solutions `k` can be specified. The function also permits finding the CNH solution of [@van2021chromosomal], which ignores the non-negativity of ACN constraint, by turning `plus` to `False`. 

### CNH

CNH estimates of the tumor purity, the tumor ploidy and the CNH value (i.e., kappa):

```{r}
# ACN profile for CNH
cnh = find_cnhplus(grid, r = r, w = w, k = 2, plus = F)
cnh
```

ACN recovered by CNH, can be obtained by `r2q` function, using the RCN data and the CNH estimates of the tumor purity, the tumor ploidy as the input.

```{r}
acn_cnh = r2q(r, cnh$purity[1], cnh$ploidy[1])
```

The ACN profile of tumor, recovered by CNH can be plotted by `plot_profile` function.

```{r, fig.width=8, fig.height=4, include=T, fig.align='center', fig.show='hold', results=FALSE}
plot_profile(sa, acn_cnh, paste0(sample_name, '_ACN_for_CNH.tiff'),
             ylim = c(-10,10), ylab = 'ACN for CNH')
```

The number of copies is negative in numerous segments.

### CNH+

By default, `find_cnhplus` searches for the CNH+ solution.

```{r, fig.width=8, fig.height=4, include=T, fig.align='center', fig.show='hold', results=FALSE}
# ACN profile for CNH+
cnh_plus = find_cnhplus(grid, r = r, w = w, k=2, plus = T)
cnh_plus
#
acn_cnhplus = r2q(r, cnh_plus$purity[1], cnh_plus$ploidy[1])
plot_profile(sa, acn_cnhplus, paste0(sample_name, '_ACN_for_CNHplus.tiff'),
             ylim = c(0,10), ylab = 'ACN for CNH+')
```

## Stratification of cancer patients for survival by CNH+

Stratification of patients from a TCGA study for survival by CNH+ can be performed using `analyze_TCGA_study` function. First, it is necessary to download CNV data from TCGA; we use `TCGAbiolinks` to download hg19-based segmented CNV data for primary tumor, obtained from Affymetrix SNP Array 6.0 platform.

```{r}
# TCGA study
study_name = 'READ'
#
query = TCGAbiolinks::GDCquery(legacy = T,
                 project = paste0('TCGA-', study_name), 
                 data.category = "Copy number variation",
                 file.type = 'hg19.seg',
                 platform = "Affymetrix SNP Array 6.0",
                 sample.type = 'Primary Tumor')
# download segmented_scna_hg19 data 
TCGAbiolinks::GDCdownload(query = query, 
            method = 'api', files.per.chunk = 10) #'client') # 
data = TCGAbiolinks::GDCprepare(query = query)
da = as.data.frame(data)
#            
```


In order to replicate analyses from [@van2021chromosomal] it is necessary to select the subset of TCGA samples that were considered by the authors. To do so, first, read the relevant data from Supplement of [@van2021chromosomal] to the `R` session.

```{r}
# read in the Supplement to van Dijk et al. with results for tumor samples from TCGA studies
vD = openxlsx::read.xlsx("https://static-content.springer.com/esm/art%3A10.1038%2Fs41467-021-23384-6/MediaObjects/41467_2021_23384_MOESM4_ESM.xlsx")
#
```

Next, select the data from the `study_name` TCGA study and match them with the CNV data.

```{r}
# filter samples from the study
vDx = vD %>% filter(Type == study_name)
#
# match vD and TCGA
im = match(vDx$Samplename, unique(da$Sample))
#
# data frame with RCN for the study samples which were considered by van Dijk et al.
da_vD = da %>% filter(Sample %in% unique(da$Sample)[im])
#
```

Analyze the TCGA study. The analysis involves finding CNH+ solution for every sample from the study. By default, only the CNH+ solutions are sought. It takes a time; even more so when both CNH+ and CNH are computed. This is why the code below is cached to save time when re-running this vignette.


```{r, cache = TRUE}
# grid has already been prepared, above
#
# analyze TCGA study
oo = analyze_TCGA_study(study_name, da_vD, grid, k=2)
#
```

The function invisibly saves csv file with the results; i.e., CNH+, purity, ploidy for each sample.

Function `plot_survival` can be used for plotting the Kaplan-Meier survival functions for two groups of samples from TCGA study: those with CNH+ below or equal to the median vs. samples with CNH+ above median.

```{r, fig.width=7, fig.height=8.5, include=T, fig.align='center', fig.show='hold'}
# read in the results file for the study
res = read.csv(paste0(study_name, '_results.csv'))
# match samples from results file and the survival data in vDx
im_vDx_res = match(vDx$Samplename, res$sample)
# KM for two groups; below/above median CNH+
gg_cnhplus = plot_survival(study_name, vDx$OS, vDx$OS_event, 
                           res$cnh_plus[im_vDx_res],
                           type = 'CNH+', ylim = c(0, 1))
gg_cnhplus
```



## References

