<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="analyze_TCGA_study takes data from a TCGA study and for each sample it finds
CNH+ and associated tumor purity, tumor ploidy. Option to find both CNH+ and CNH is provided."><title>CNH+ and associated tumor purity, tumor ploidy for samples from TCGA study are found — analyze_TCGA_study • CNHplus</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="CNH+ and associated tumor purity, tumor ploidy for samples from TCGA study are found — analyze_TCGA_study"><meta property="og:description" content="analyze_TCGA_study takes data from a TCGA study and for each sample it finds
CNH+ and associated tumor purity, tumor ploidy. Option to find both CNH+ and CNH is provided."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CNHplus</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/CNHplus.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/grendar/CNHplus/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>CNH+ and associated tumor purity, tumor ploidy for samples from TCGA study are found</h1>
      <small class="dont-index">Source: <a href="https://github.com/grendar/CNHplus/blob/HEAD/R/analyze_TCGA_study.R" class="external-link"><code>R/analyze_TCGA_study.R</code></a></small>
      <div class="d-none name"><code>analyze_TCGA_study.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>analyze_TCGA_study</code> takes data from a TCGA study and for each sample it finds
CNH+ and associated tumor purity, tumor ploidy. Option to find both CNH+ and CNH is provided.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">analyze_TCGA_study</span><span class="op">(</span><span class="va">study_name</span>, <span class="va">da</span>, <span class="va">grid</span>, <span class="va">k</span>, both <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>study_name</dt>
<dd><p>TCGA study name</p></dd>


<dt>da</dt>
<dd><p>data frame containing Sample, Chromosome, Start, End, Num_Probes, Segment_Mean</p></dd>


<dt>grid</dt>
<dd><p>grid of purities, ploidies over which to search for CNH+ (matrix)</p></dd>


<dt>k</dt>
<dd><p>how many candidate solution to return</p></dd>


<dt>both</dt>
<dd><p>whether to find CNH+ as well as CNH (default is F, i.e., CNH+ only)</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>saves csv results file named after the study, with CNH+, purity, ploidy for each sample</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Example shows how to
1) download RCN profiles for samples from a TCGA study;
2) read in a Supplementary file from van Dijk et al. which contains sample names
   of the TCGA tumor samples that were analyzed by the authors, as well as other data
   (CNH, survival data);
3) select from the downloaded TCGA RCN data only those samples that were analyzed
   by van Dijk et al.;
4) make grid of purities, ploidies over which the CNH+ solution will be searched for;
5) analyze the TCGA study - for each samples find CNH+ and associated purity, ploidy pair
6) compare survival curves of subjects with CNH+ below/above median and make KM plot</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>van Dijk E, van den Bosch T, Lenos KJ, El Makrini K, Nijman LE, van Essen HF, Lansu N, Boekhout M, Hageman JH, Fitzgerald RC, others (2021).
“<a href="https://doi.org/10.1038/s41467-021-23384-6" class="external-link">Chromosomal copy number heterogeneity predicts survival rates across cancers</a>.”
<em>Nature communications</em>, <b>12</b>(1), 1--12.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="co"># Ex:</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/BioinformaticsFMRP/TCGAbiolinks" class="external-link">'TCGAbiolinks'</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://stringi.gagolewski.com/" class="external-link">'stringi'</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ycphs.github.io/openxlsx/index.html" class="external-link">'openxlsx'</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org" class="external-link">'dplyr'</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">#</span></span></span>
<span class="r-in"><span><span class="co"># TCGA study</span></span></span>
<span class="r-in"><span><span class="va">study_name</span> <span class="op">=</span> <span class="st">'READ'</span></span></span>
<span class="r-in"><span><span class="co">#</span></span></span>
<span class="r-in"><span><span class="va">query</span> <span class="op">=</span> <span class="fu">TCGAbiolinks</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TCGAbiolinks/man/GDCquery.html" class="external-link">GDCquery</a></span><span class="op">(</span>legacy <span class="op">=</span> <span class="cn">T</span>,</span></span>
<span class="r-in"><span>                               project <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'TCGA-'</span>, <span class="va">study_name</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                               data.category <span class="op">=</span> <span class="st">"Copy number variation"</span>,</span></span>
<span class="r-in"><span>                               file.type <span class="op">=</span> <span class="st">'hg19.seg'</span>,</span></span>
<span class="r-in"><span>                               platform <span class="op">=</span> <span class="st">"Affymetrix SNP Array 6.0"</span>,</span></span>
<span class="r-in"><span>                               sample.type <span class="op">=</span> <span class="st">'Primary Tumor'</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># download segmented_scna_hg19 data</span></span></span>
<span class="r-in"><span><span class="fu">TCGAbiolinks</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TCGAbiolinks/man/GDCdownload.html" class="external-link">GDCdownload</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">query</span>,</span></span>
<span class="r-in"><span>                          method <span class="op">=</span> <span class="st">"api"</span>, files.per.chunk <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="co"># client')</span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">=</span> <span class="fu">TCGAbiolinks</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TCGAbiolinks/man/GDCprepare.html" class="external-link">GDCprepare</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">query</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">da</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">#</span></span></span>
<span class="r-in"><span><span class="co"># read in the Supplement to van Dijk et al. with results for tumor samples from TCGA studies</span></span></span>
<span class="r-in"><span><span class="va">vD</span> <span class="op">=</span> <span class="fu">openxlsx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/openxlsx/man/read.xlsx.html" class="external-link">read.xlsx</a></span><span class="op">(</span><span class="st">"https://static-content.springer.com/esm/art%3A10.1038%2Fs41467-021-23384-6/MediaObjects/41467_2021_23384_MOESM4_ESM.xlsx"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">#</span></span></span>
<span class="r-in"><span><span class="co"># filter samples from the study</span></span></span>
<span class="r-in"><span><span class="va">vDx</span> <span class="op">=</span> <span class="va">vD</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Type</span> <span class="op">==</span> <span class="va">study_name</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">#</span></span></span>
<span class="r-in"><span><span class="co"># match vD and TCGA</span></span></span>
<span class="r-in"><span><span class="va">im</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">vDx</span><span class="op">$</span><span class="va">Samplename</span>, <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">da</span><span class="op">$</span><span class="va">Sample</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">#</span></span></span>
<span class="r-in"><span><span class="co"># data frame with RCN for the study samples which were considered by van Dijk et al.</span></span></span>
<span class="r-in"><span><span class="va">da_vD</span> <span class="op">=</span> <span class="va">da</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">da</span><span class="op">$</span><span class="va">Sample</span><span class="op">)</span><span class="op">[</span><span class="va">im</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">#</span></span></span>
<span class="r-in"><span><span class="co"># grid</span></span></span>
<span class="r-in"><span><span class="va">grid</span> <span class="op">=</span> <span class="fu"><a href="make_grid.html">make_grid</a></span><span class="op">(</span>purity <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">1</span>, <span class="fl">0.01</span><span class="op">)</span>, ploidy <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1.5</span>, <span class="fl">5</span>, <span class="fl">0.01</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">#</span></span></span>
<span class="r-in"><span><span class="co"># analyze TCGA study</span></span></span>
<span class="r-in"><span><span class="va">oo</span> <span class="op">=</span> <span class="fu">analyze_TCGA_study</span><span class="op">(</span><span class="va">study_name</span>, <span class="va">da_vD</span>, <span class="va">grid</span>, k<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">study_name</span>, <span class="st">'_results.csv'</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># match samples from results file and the survival data in vDx</span></span></span>
<span class="r-in"><span><span class="va">im_vDx_res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">vDx</span><span class="op">$</span><span class="va">Samplename</span>, <span class="va">res</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">#</span></span></span>
<span class="r-in"><span><span class="co"># survival analysis (below/above median CNH+)</span></span></span>
<span class="r-in"><span><span class="va">gg_cnhplus</span> <span class="op">=</span> <span class="fu"><a href="plot_survival.html">plot_survival</a></span><span class="op">(</span><span class="va">study_name</span>,</span></span>
<span class="r-in"><span>                           <span class="va">vDx</span><span class="op">$</span><span class="va">OS</span>, <span class="va">vDx</span><span class="op">$</span><span class="va">OS_event</span>,</span></span>
<span class="r-in"><span>                           <span class="va">res</span><span class="op">$</span><span class="va">cnh_plus</span><span class="op">[</span><span class="va">im_vDx_survival</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>                           type <span class="op">=</span> <span class="st">'unfiltered CNH+'</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gg_cnhplus</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Marian Grendar.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

