<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="CNHplus">
<title>CNHplus: the chromosomal copy number heterogeneity which respects biological constraints • CNHplus</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="CNHplus: the chromosomal copy number heterogeneity which respects biological constraints">
<meta property="og:description" content="CNHplus">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CNHplus</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/CNHplus.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/grendar/CNHplus/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>CNHplus: the chromosomal copy number heterogeneity which respects biological constraints</h1>
                        <h4 data-toc-skip class="author">Marian Grendár</h4>
            
            <h4 data-toc-skip class="date">05/10/2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/grendar/CNHplus/blob/HEAD/vignettes/CNHplus.Rmd" class="external-link"><code>vignettes/CNHplus.Rmd</code></a></small>
      <div class="d-none name"><code>CNHplus.Rmd</code></div>
    </div>

    
    
<style type="text/css">
  body{
  font-size: 12pt;
   font-family: "Times New Roman", Times, serif;
}
</style>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><em>CNHplus</em> provides R functions for estimating the tumor purity, the tumor ploidy, the Absolute Copy Number (ACN) profile of a tumor and the Copy Number Heterogeneity (CNH) <span class="citation">(van Dijk et al. 2021)</span> from the Relative Copy Number (RCN) profile of a sample, by the CNHplus method <span class="citation">(Grendár et al. 2022)</span>. CNHplus modifies the original CNH method <span class="citation">(van Dijk et al. 2021)</span> by imposing the non-negativity constraint on the recovered ACN profile.</p>
</div>
<div class="section level2">
<h2 id="cnh-and-cnh">CNH and CNH+<a class="anchor" aria-label="anchor" href="#cnh-and-cnh"></a>
</h2>
<p><span class="citation">(van Dijk et al. 2021)</span> introduced a single-sample method for estimating the Intra-Tumor Heterogeneity from Copy Number Variation (CNV) data, dubbed Copy Number Heterogeneity. The ACN profiles, recovered by CNH, may contain segments with the negative number of copies; cf. <span class="citation">(Grendár et al. 2022)</span>. CNHplus method, - CNH+, for short, - implemented in <em>CNHplus</em> library, corrects this omission in CNH as it imposes the non-negativity constraint on ACN of tumor in searching for the solution of the optimization problem that defines CNH.</p>
</div>
<div class="section level2">
<h2 id="loading-cnhplus">Loading CNHplus<a class="anchor" aria-label="anchor" href="#loading-cnhplus"></a>
</h2>
<p><em>CNHplus</em> and other libraries necessary for illustrating its use should be loaded, at first.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/grendar/CNHplus" class="external-link">'CNHplus'</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/BioinformaticsFMRP/TCGAbiolinks" class="external-link">'TCGAbiolinks'</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://stringi.gagolewski.com/" class="external-link">'stringi'</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ycphs.github.io/openxlsx/index.html" class="external-link">'openxlsx'</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org" class="external-link">'dplyr'</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="input-data">Input data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h2>
<p>The RCN profile of a sample, which is the input to CNH computations, can be obtained either by the microarray technology or by shallow Whole Genome Sequencing. The RCN profile data frame should comprise the following variables, at minimum:</p>
<ul>
<li><code>Sample</code></li>
<li><code>Chromosome</code></li>
<li><code>Start</code></li>
<li><code>End</code></li>
<li><code>Segment_Mean</code></li>
</ul>
<p>Note that <em>CNHplus</em> assumes that Segment_Mean is the relative copy number of the segment, not its log2 transformation.</p>
</div>
<div class="section level2">
<h2 id="plotting-rcn-profile-of-a-sample">Plotting RCN profile of a sample<a class="anchor" aria-label="anchor" href="#plotting-rcn-profile-of-a-sample"></a>
</h2>
<p>In order to reproduce the example from <span class="citation">(Grendár et al. 2022)</span>, it is necessary to load the RCN data for the TCGA-AJ-A3EJ-01A-11D-A19X-01 sample, which are included in <em>CNHplus</em>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">sample_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sample_data</span><span class="op">)</span></span>
<span><span class="co">#&gt;                         Sample Chromosome    Start      End Num_Probes</span></span>
<span><span class="co">#&gt; 1 TCGA-AJ-A3EJ-01A-11D-A19X-01          1    61735   356542         29</span></span>
<span><span class="co">#&gt; 2 TCGA-AJ-A3EJ-01A-11D-A19X-01          1   404171 15599807       8074</span></span>
<span><span class="co">#&gt; 3 TCGA-AJ-A3EJ-01A-11D-A19X-01          1 15600049 16065607        266</span></span>
<span><span class="co">#&gt; 4 TCGA-AJ-A3EJ-01A-11D-A19X-01          1 16075128 16153497         54</span></span>
<span><span class="co">#&gt; 5 TCGA-AJ-A3EJ-01A-11D-A19X-01          1 16153536 16155010          9</span></span>
<span><span class="co">#&gt; 6 TCGA-AJ-A3EJ-01A-11D-A19X-01          1 16165661 16830820        295</span></span>
<span><span class="co">#&gt;   Segment_Mean</span></span>
<span><span class="co">#&gt; 1      -0.6532</span></span>
<span><span class="co">#&gt; 2       0.1152</span></span>
<span><span class="co">#&gt; 3       0.4350</span></span>
<span><span class="co">#&gt; 4       0.8853</span></span>
<span><span class="co">#&gt; 5      -1.4492</span></span>
<span><span class="co">#&gt; 6       0.9010</span></span></code></pre></div>
<p>Next, in order to plot the RCN profile of the sample, it is necessary to prepare the RCN profile (by anti-log2 transforming the Segment_Mean data) and a vector with the widths of the segments.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sa</span> <span class="op">=</span> <span class="va">sample_data</span></span>
<span><span class="va">sample_name</span> <span class="op">=</span> <span class="va">sa</span><span class="op">$</span><span class="va">Sample</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">r</span> <span class="op">=</span> <span class="fl">2</span><span class="op">^</span><span class="va">sa</span><span class="op">$</span><span class="va">Segment_Mean</span></span>
<span><span class="va">w</span> <span class="op">=</span> <span class="va">sa</span><span class="op">$</span><span class="va">End</span> <span class="op">-</span> <span class="va">sa</span><span class="op">$</span><span class="va">Start</span> <span class="co"># should be +1</span></span></code></pre></div>
<p><code>plot_profile</code> function then plots the RCN profile of the sample and saves it to a tiff file with file name specified by <code>sample_name</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># RCN profile plot</span></span>
<span><span class="fu"><a href="../reference/plot_profile.html">plot_profile</a></span><span class="op">(</span><span class="va">sa</span>, <span class="va">r</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">sample_name</span>, <span class="st">'_RCN.tiff'</span><span class="op">)</span>,</span>
<span>             ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">5</span><span class="op">)</span>, ylab <span class="op">=</span> <span class="st">'RCN'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CNHplus_files/figure-html/unnamed-chunk-4-1.png" width="768" style="display: block; margin: auto;"></p>
<p>The function can be used also for plotting the ACN profiles.</p>
</div>
<div class="section level2">
<h2 id="finding-cnh-and-cnh">Finding CNH+ (and CNH)<a class="anchor" aria-label="anchor" href="#finding-cnh-and-cnh"></a>
</h2>
<p><em>CNHplus</em> finds a solution of the optimization problem (4’) from <span class="citation">(Grendár et al. 2022)</span> by searching over a grid of pairs of tumor purities and tumor ploidies. The grid can be created by <code>make_grid</code> function.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># making grid</span></span>
<span><span class="va">grid</span> <span class="op">=</span> <span class="fu"><a href="../reference/make_grid.html">make_grid</a></span><span class="op">(</span>purity <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">1</span>, <span class="fl">0.01</span><span class="op">)</span>, ploidy <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1.5</span>, <span class="fl">5</span>, <span class="fl">0.01</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><code>find_cnhplus</code> function requires the grid and relative copy number profile as input. The number of candidate solutions <code>k</code> can be specified. The function also permits finding the CNH solution of <span class="citation">(van Dijk et al. 2021)</span>, which ignores the non-negativity of ACN constraint, by turning <code>plus</code> to <code>False</code>.</p>
<div class="section level3">
<h3 id="cnh">CNH<a class="anchor" aria-label="anchor" href="#cnh"></a>
</h3>
<p>CNH estimates of the tumor purity, the tumor ploidy and the CNH value (i.e., kappa):</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ACN profile for CNH</span></span>
<span><span class="va">cnh</span> <span class="op">=</span> <span class="fu"><a href="../reference/find_cnhplus.html">find_cnhplus</a></span><span class="op">(</span><span class="va">grid</span>, r <span class="op">=</span> <span class="va">r</span>, w <span class="op">=</span> <span class="va">w</span>, k <span class="op">=</span> <span class="fl">2</span>, plus <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="va">cnh</span></span>
<span><span class="co">#&gt;   purity ploidy     kappa</span></span>
<span><span class="co">#&gt; 1   0.24   2.87 0.1384375</span></span>
<span><span class="co">#&gt; 2   0.27   3.86 0.1387208</span></span></code></pre></div>
<p>ACN recovered by CNH, can be obtained by <code>r2q</code> function, using the RCN data and the CNH estimates of the tumor purity, the tumor ploidy as the input.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">acn_cnh</span> <span class="op">=</span> <span class="fu"><a href="../reference/r2q.html">r2q</a></span><span class="op">(</span><span class="va">r</span>, <span class="va">cnh</span><span class="op">$</span><span class="va">purity</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="va">cnh</span><span class="op">$</span><span class="va">ploidy</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p>The ACN profile of tumor, recovered by CNH can be plotted by <code>plot_profile</code> function.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_profile.html">plot_profile</a></span><span class="op">(</span><span class="va">sa</span>, <span class="va">acn_cnh</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">sample_name</span>, <span class="st">'_ACN_for_CNH.tiff'</span><span class="op">)</span>,</span>
<span>             ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">10</span>,<span class="fl">10</span><span class="op">)</span>, ylab <span class="op">=</span> <span class="st">'ACN for CNH'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CNHplus_files/figure-html/unnamed-chunk-8-1.png" width="768" style="display: block; margin: auto;"></p>
<p>The number of copies is negative in numerous segments.</p>
</div>
<div class="section level3">
<h3 id="cnh-1">CNH+<a class="anchor" aria-label="anchor" href="#cnh-1"></a>
</h3>
<p>By default, <code>find_cnhplus</code> searches for the CNH+ solution.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ACN profile for CNH+</span></span>
<span><span class="va">cnh_plus</span> <span class="op">=</span> <span class="fu"><a href="../reference/find_cnhplus.html">find_cnhplus</a></span><span class="op">(</span><span class="va">grid</span>, r <span class="op">=</span> <span class="va">r</span>, w <span class="op">=</span> <span class="va">w</span>, k<span class="op">=</span><span class="fl">2</span>, plus <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">cnh_plus</span></span>
<span><span class="co">#</span></span>
<span><span class="va">acn_cnhplus</span> <span class="op">=</span> <span class="fu"><a href="../reference/r2q.html">r2q</a></span><span class="op">(</span><span class="va">r</span>, <span class="va">cnh_plus</span><span class="op">$</span><span class="va">purity</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="va">cnh_plus</span><span class="op">$</span><span class="va">ploidy</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_profile.html">plot_profile</a></span><span class="op">(</span><span class="va">sa</span>, <span class="va">acn_cnhplus</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">sample_name</span>, <span class="st">'_ACN_for_CNHplus.tiff'</span><span class="op">)</span>,</span>
<span>             ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">10</span><span class="op">)</span>, ylab <span class="op">=</span> <span class="st">'ACN for CNH+'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CNHplus_files/figure-html/unnamed-chunk-9-1.png" width="768" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="stratification-of-cancer-patients-for-survival-by-cnh">Stratification of cancer patients for survival by CNH+<a class="anchor" aria-label="anchor" href="#stratification-of-cancer-patients-for-survival-by-cnh"></a>
</h2>
<p>Stratification of patients from a TCGA study for survival by CNH+ can be performed using <code>analyze_TCGA_study</code> function. First, it is necessary to download CNV data from TCGA; we use <code>TCGAbiolinks</code> to download hg19-based segmented CNV data for primary tumor, obtained from Affymetrix SNP Array 6.0 platform.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># TCGA study</span></span>
<span><span class="va">study_name</span> <span class="op">=</span> <span class="st">'READ'</span></span>
<span><span class="co">#</span></span>
<span><span class="va">query</span> <span class="op">=</span> <span class="fu">TCGAbiolinks</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TCGAbiolinks/man/GDCquery.html" class="external-link">GDCquery</a></span><span class="op">(</span>legacy <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                 project <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'TCGA-'</span>, <span class="va">study_name</span><span class="op">)</span>, </span>
<span>                 data.category <span class="op">=</span> <span class="st">"Copy number variation"</span>,</span>
<span>                 file.type <span class="op">=</span> <span class="st">'hg19.seg'</span>,</span>
<span>                 platform <span class="op">=</span> <span class="st">"Affymetrix SNP Array 6.0"</span>,</span>
<span>                 sample.type <span class="op">=</span> <span class="st">'Primary Tumor'</span><span class="op">)</span></span>
<span><span class="co">#&gt; --------------------------------------</span></span>
<span><span class="co">#&gt; o GDCquery: Searching in GDC database</span></span>
<span><span class="co">#&gt; --------------------------------------</span></span>
<span><span class="co">#&gt; Genome of reference: hg19</span></span>
<span><span class="co">#&gt; --------------------------------------------</span></span>
<span><span class="co">#&gt; oo Accessing GDC. This might take a while...</span></span>
<span><span class="co">#&gt; --------------------------------------------</span></span>
<span><span class="co">#&gt; ooo Project: TCGA-READ</span></span>
<span><span class="co">#&gt; --------------------</span></span>
<span><span class="co">#&gt; oo Filtering results</span></span>
<span><span class="co">#&gt; --------------------</span></span>
<span><span class="co">#&gt; ooo By platform</span></span>
<span><span class="co">#&gt; ooo By file.type</span></span>
<span><span class="co">#&gt; ooo By sample.type</span></span>
<span><span class="co">#&gt; ----------------</span></span>
<span><span class="co">#&gt; oo Checking data</span></span>
<span><span class="co">#&gt; ----------------</span></span>
<span><span class="co">#&gt; ooo Checking if there are duplicated cases</span></span>
<span><span class="co">#&gt; ooo Checking if there are results for the query</span></span>
<span><span class="co">#&gt; -------------------</span></span>
<span><span class="co">#&gt; o Preparing output</span></span>
<span><span class="co">#&gt; -------------------</span></span>
<span><span class="co"># download segmented_scna_hg19 data </span></span>
<span><span class="fu">TCGAbiolinks</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TCGAbiolinks/man/GDCdownload.html" class="external-link">GDCdownload</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">query</span>, </span>
<span>            method <span class="op">=</span> <span class="st">'api'</span>, files.per.chunk <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="co">#'client') # </span></span>
<span><span class="co">#&gt; Downloading data for project TCGA-READ</span></span>
<span><span class="co">#&gt; Of the 169 files for download 169 already exist.</span></span>
<span><span class="co">#&gt; All samples have been already downloaded</span></span>
<span><span class="va">data</span> <span class="op">=</span> <span class="fu">TCGAbiolinks</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TCGAbiolinks/man/GDCprepare.html" class="external-link">GDCprepare</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">query</span><span class="op">)</span></span>
<span><span class="co">#&gt; Reading copy number variation files</span></span>
<span><span class="va">da</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span>
<span><span class="co">#            </span></span></code></pre></div>
<p>In order to replicate analyses from <span class="citation">(van Dijk et al. 2021)</span> it is necessary to select the subset of TCGA samples that were considered by the authors. To do so, first, read the relevant data from Supplement of <span class="citation">(van Dijk et al. 2021)</span> to the <code>R</code> session.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># read in the Supplement to van Dijk et al. with results for tumor samples from TCGA studies</span></span>
<span><span class="va">vD</span> <span class="op">=</span> <span class="fu">openxlsx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/openxlsx/man/read.xlsx.html" class="external-link">read.xlsx</a></span><span class="op">(</span><span class="st">"https://static-content.springer.com/esm/art%3A10.1038%2Fs41467-021-23384-6/MediaObjects/41467_2021_23384_MOESM4_ESM.xlsx"</span><span class="op">)</span></span>
<span><span class="co">#</span></span></code></pre></div>
<p>Next, select the data from the <code>study_name</code> TCGA study and match them with the CNV data.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># filter samples from the study</span></span>
<span><span class="va">vDx</span> <span class="op">=</span> <span class="va">vD</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Type</span> <span class="op">==</span> <span class="va">study_name</span><span class="op">)</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># match vD and TCGA</span></span>
<span><span class="va">im</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">vDx</span><span class="op">$</span><span class="va">Samplename</span>, <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">da</span><span class="op">$</span><span class="va">Sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># data frame with RCN for the study samples which were considered by van Dijk et al.</span></span>
<span><span class="va">da_vD</span> <span class="op">=</span> <span class="va">da</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">da</span><span class="op">$</span><span class="va">Sample</span><span class="op">)</span><span class="op">[</span><span class="va">im</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#</span></span></code></pre></div>
<p>Analyze the TCGA study. The analysis involves finding CNH+ solution for every sample from the study. By default, only the CNH+ solutions are sought. It takes a time; even more so when both CNH+ and CNH are computed. This is why the code below is cached to save time when re-running this vignette.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># grid has already been prepared, above</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># analyze TCGA study</span></span>
<span><span class="va">oo</span> <span class="op">=</span> <span class="fu"><a href="../reference/analyze_TCGA_study.html">analyze_TCGA_study</a></span><span class="op">(</span><span class="va">study_name</span>, <span class="va">da_vD</span>, <span class="va">grid</span>, k<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#</span></span></code></pre></div>
<p>The function invisibly saves csv file with the results; i.e., CNH+, purity, ploidy for each sample.</p>
<p>Function <code>plot_survival</code> can be used for plotting the Kaplan-Meier survival functions for two groups of samples from TCGA study: those with CNH+ below or equal to the median vs. samples with CNH+ above median.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># read in the results file for the study</span></span>
<span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">study_name</span>, <span class="st">'_results.csv'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># match samples from results file and the survival data in vDx</span></span>
<span><span class="va">im_vDx_res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">vDx</span><span class="op">$</span><span class="va">Samplename</span>, <span class="va">res</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="co"># KM for two groups; below/above median CNH+</span></span>
<span><span class="va">gg_cnhplus</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_survival.html">plot_survival</a></span><span class="op">(</span><span class="va">study_name</span>, <span class="va">vDx</span><span class="op">$</span><span class="va">OS</span>, <span class="va">vDx</span><span class="op">$</span><span class="va">OS_event</span>, </span>
<span>                           <span class="va">res</span><span class="op">$</span><span class="va">cnh_plus</span><span class="op">[</span><span class="va">im_vDx_res</span><span class="op">]</span>,</span>
<span>                           type <span class="op">=</span> <span class="st">'CNH+'</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">gg_cnhplus</span></span></code></pre></div>
<p><img src="CNHplus_files/figure-html/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-grendar2022cnhplus" class="csl-entry">
Grendár, Marian, Petr Martı́nek, Dušan Loderer, and Ondrej Ondič. 2022. <span>“CNHplus: The Chromosomal Copy Number Heterogeneity Which Respects Biological Constraints.”</span> <em>bioRxiv</em>. <a href="https://doi.org/10.1101/2022.09.30.510279" class="external-link">https://doi.org/10.1101/2022.09.30.510279</a>.
</div>
<div id="ref-van2021chromosomal" class="csl-entry">
van Dijk, Erik, Tom Van Den Bosch, Kristiaan J Lenos, Khalid El Makrini, Lisanne E Nijman, Hendrik FB van Essen, Nico Lansu, et al. 2021. <span>“<a href="https://doi.org/10.1038/s41467-021-23384-6" class="external-link">Chromosomal Copy Number Heterogeneity Predicts Survival Rates Across Cancers</a>.”</span> <em>Nature Communications</em> 12 (1): 1–12.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Marian Grendar.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
